cachix/src/Cachix/Deploy/Lock.hs

```diff
@@ -0,0 +1,56 @@
+module Cachix.Deploy.Lock (withTryLock) where
+
+import qualified Lukko as Lock
+import Protolude hiding ((<.>))
+import qualified System.Directory as Directory
+import System.FilePath ((<.>), (</>))
+import qualified System.FilePath as FilePath
+import qualified System.Posix.Files as Files
+import qualified System.Posix.Types as Posix
+
+defaultLockDirectory :: FilePath
+defaultLockDirectory = "cachix" </> "deploy" </> "locks"
+
+-- | Get the path to the lock directory based on the ownership of the profile.
+--
+-- For user-owned profiles: $XDG_CACHE_DIR/cachix/deploy/locks
+-- For root-owned profiles: /var/run/cachix/deploy/locks
+getLockDirectoryFromProfile :: FilePath -> IO FilePath
+getLockDirectoryFromProfile profilePath = do
+  userId <- Files.fileOwner <$> Files.getFileStatus (FilePath.takeDirectory profilePath)
+  if isRoot userId
+    then pure $ "/var/run" </> defaultLockDirectory
```

Comment:
In order to minimize burden of support, it's best to always use the same path regardless of the user.

If we don't specialize `/var/run` and just use xdg's `/root/.cache` case that should work well, what do you think?

Reply:
I had no idea root had it’s own little space there 😂 That‘s awesome!

What happens on macOS though? Running `getXdgDirectory` with sudo, it returns the user cache directory and I end up creating a folder owned by root.

Reply:
That might be because you need `sudo -H`, here's a session from m1 macos:

```
Prelude> import qualified System.Directory as Directory
Prelude Directory> Directory.getXdgDirectory Directory.XdgCache ""
"/var/root/.cache"
Prelude Directory> 
```

Reply:
Ah, this again. I assume this works when running via launchd, but do we do if the user runs the agent manually? Show a warning if `$HOME` is incorrect?

Reply:
Good question, a warning seems like a good idea.